src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -122,6 +122,62 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
 
   Blazor._internal.forceCloseConnection = () => connection.stop();
 
+  Blazor._internal.sendJSDataStream = (data: ArrayBufferView, streamId: string, maximumIncomingBytes: number) => {
+    // Run the rest in the background, without delaying the completion of the call to sendJSDataStream
+    // otherwise we'll deadlock (.NET can't begin reading until this completes, but it won't complete
+    // because nobody's reading the pipe)
+    setTimeout(async () => {
+      const maxMillisecondsBetweenAcks = 500;
+      let numChunksUntilNextAck = 5;
```

Comment:
I think we might want these to be configured by the app developer on the call to OpenRead, since this is something that affects the user experience.

* How does reducing/removing the number of milliseconds between acks affects the upload speed?
* How does the number of chunks until the next ack affect the speed?

If we determine that below 500ms and 5 chunks we don't get speed increases, independent of the machine speed) then I think the defaults are fine, that said, we might want to flow them from the server where they can at least be globally configured.

If we determine that by avoiding limits (at the cost of responsiveness) we can get significant speed updates, I think we should make these configurable on a per call basis by the developer.

Here are two scenarios:
* A developer uploading a huge amount of data where they want to go "as fast as they can" at the cost of UI responsiveness. (For example if they just display a UI with updates and no options to interact with the page in that state).
* A developer uploading a single file, willing to pay the one time cost of the responsiveness hit when a form is submitted.

Reply:
If we want to make this configurable, then I think it should be done in terms of the `maxMillisecondsBetweenAcks`, and *not* in terms of `numChunksUntilNextAck` (since this is computed from `maxMillisecondsBetweenAcks` as soon as the data starts flowing).

However:

 * We don't have to do that in the first version. We could wait for more concrete customer requests before deciding there's really a role for making this configurable.
 * There are legit reasons to think that all applications should pick roughly the same value, because this is about how much lagginess you're willing to add to interactivity, which is a question about how much humans are willing to wait. It's quite possible that everyone will be roughly happy with 500ms as the max duration for delaying the reaction to some UI interaction. Nobody would want to pick (say) < 10ms or > 5s, and even if you did it wouldn't be *that* transformative to the upload speed.

So personally I'd be happy to leave this fixed at 500ms for the initial release and then we can see if customers actually seem interested in practice in changing that.